Author: Mike <m.turner14@imperial.ac.uk>
Description: use size_t for tagIndex instead of int (and casting tag->length)
  https://gitlab.nektar.info/nektar/nektar/-/blob/master/cmake/scripts/tinyxml.patch
--- tinyxmlparser.cpp	2017-10-03 16:05:35.999295200 +0100
+++ tinyxmlparser.cpp	2017-10-03 16:05:58.556565300 +0100
@@ -655,7 +655,7 @@
 
 	while ( in->good() )
 	{
-		int tagIndex = (int) tag->length();
+		size_t tagIndex = tag->length();
 		while ( in->good() && in->peek() != '>' )
 		{
 			int c = in->get();
@@ -958,7 +958,7 @@
 			// We should be at a "<", regardless.
 			if ( !in->good() ) return;
 			assert( in->peek() == '<' );
-			int tagIndex = (int) tag->length();
+			size_t tagIndex = tag->length();
 
 			bool closingTag = false;
 			bool firstCharFound = false;
